src/components/IPFSHOC/index.tsx

```diff
@@ -1,128 +1,77 @@
+        const { ipfsGetAndDecode } = await promisedIPFS
 
-        await checkTokenListJSON(json)
+        this.setState({ pullingData: true, error: null })
 
-        // HTML5 API to read file and set state as contents
-        const fileBuffer = await readFileUpload(oFile)
+        const fileContent: any = await Promise.race([
+          ipfsGetAndDecode(fileHash),
+          timeoutCondition(IPFS_TIMEOUT, 'IPFS timeout, please check hash and try again'),
+        ])
 
-        // setState
-        return setUploadFileParams({ oFile, fileBuffer, json })
-      } catch (error) {
-        console.error(error)
-        return openModal({
-          modalName: 'TransactionModal',
-          modalProps: {
-            header: `File Upload Error`,
-            body: `
-            Please check that you correctly selected a valid JSON file less than 1MB.
-            File must be formatted specifically as discussed here: https://some-site.com
-            `,
-            button: true,
-            error: error.message || 'Unknown error occurred, please open your developer console',
-          },
-        })
-      }
+        const json = JSON.parse(fileContent)
 
-    }
+        await checkTokenListJSON(json)
 
-    handleSendToIPFS = async () => {
-      const { ipfsAddFile } = await promisedIPFS
-      const { fileBuffer, oFile, json, setNewIPFSCustomListAndUpdateBalances, openModal } = this.props
-      try {
-        const { fileHash, filePath } = await ipfsAddFile(fileBuffer, oFile),
-          customTokenListWithDecimals = await getAllTokenDecimals(json)
+        const customTokenListWithDecimals = await getAllTokenDecimals(json)
 
         localForage.setItem('customListHash', fileHash)
+        localForage.setItem('customTokenList', customTokenListWithDecimals)
 
-        // setState
-        return setNewIPFSCustomListAndUpdateBalances(
-          {
-            customTokenList: customTokenListWithDecimals,
-          },
-          {
-            fileHash,
-            filePath,
-          },
-        )
-      } catch (error) {
-        console.error(error)
-        return openModal({
-          modalName: 'TransactionModal',
-          modalProps: {
-            header: `IPFS Send Error`,
-            button: true,
-            error: error.message || 'Unknown error occurred, please open your developer console',
-          },
-        })
-      }
-    }
-
-    handleGrabFromIPFS = async () => {
-      const { ipfsGetAndDecode } = await promisedIPFS, { fileHash, getFileContentFromIPFS, openModal } = this.props
-      try {
-        const fileContent = await ipfsGetAndDecode(fileHash)
+        this.setState({ pullingData: false })
 
         // setState
-        return getFileContentFromIPFS({ fileContent })
+        return setNewIPFSCustomListAndUpdateBalances({ customTokenList: customTokenListWithDecimals })
       } catch (error) {
         console.error(error)
+        // reset hash to empty string
+        setIPFSFileHash('')
+
```

Comment:
you never change state on caught error. Like `this.setState({ pullingData: false, error })`